Input file thumbnail size doesn't change when zooming with Cmd+ or Cmd-

rr-codes opened the pull request:
#### cdb7be107d145a1b61f4d939ec3b1d4aa2fa4a16
<pre>
Input file thumbnail size doesn&apos;t change when zooming with Cmd+ or Cmd-
<a href="https://bugs.webkit.org/show_bug.cgi?id=247638">https://bugs.webkit.org/show_bug.cgi?id=247638</a>
rdar://102385998

Reviewed by Aditya Keerthi.

This PR just removes some `IOS_FAMILY` guards so that macOS behavior matches iOS.

* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::nodeHeight):
(WebCore::RenderFileUploadControl::maxFilenameWidth const):
(WebCore::RenderFileUploadControl::paintControl):

Canonical link: <a href="https://commits.webkit.org/258277@main">https://commits.webkit.org/258277@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/90a46f56f560d96eda73466502ac9cb864fe5ddb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101370 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10528 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110643 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170914 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105351 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11478 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1382 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93766 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108472 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107152 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8725 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91967 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35252 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90623 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23374 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78249 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4145 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24886 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4197 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1307 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10295 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44372 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5692 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5959 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (commented):
Rather than removing the platform conditionals entirely, I think we should guard the behavior change with `PLATFORM(COCOA)`, since we don't know how this will affect other ports.

Especially because `RenderTheme::paintFileUploadIconDecorations` is only called on `PLATFORM(COCOA)`.

pxlcoder commented on Source/WebCore/rendering/RenderFileUploadControl.cpp:
```diff
@@ -46,18 +46,22 @@ using namespace HTMLNames;
-const int iconFilenameSpacing = afterButtonSpacing;
-const int defaultWidthNumChars = 38;
+constexpr int iconFilenameSpacing = afterButtonSpacing;
+constexpr int defaultWidthNumChars = 38;
```
I think this would be easier to read if it was structured as:

```
#if PLATFORM(COCOA)
#if PLATFORM(IOS_FAMILY)
...
#else
...
#endif 
#else
...
#endif
```

rr-codes replied:
@pxlcoder that would result in
```
#if PLATFORM(COCOA)
#if PLATFORM(IOS_FAMILY)
// On iOS the icon height matches the button height, to maximize the icon size.
constexpr int iconFilenameSpacing = afterButtonSpacing;
constexpr int defaultWidthNumChars = 38;
#else
constexpr int iconFilenameSpacing = 2;
constexpr int defaultWidthNumChars = 34;
#endif
#else
constexpr int iconHeight = 16;
constexpr int iconWidth = 16;
constexpr int iconFilenameSpacing = 2;
constexpr int defaultWidthNumChars = 34;
#endif
```

which I personally think is more confusing, plus it adds some duplication

pxlcoder replied:
Sounds good. Let's just move the comment "// On iOS the icon height matches the button height, to maximize the icon size." underneath line 52, and reword as "On Cocoa platforms...", since it's out of place now.

rr-codes committed:
Input file thumbnail size doesn't change when zooming with Cmd+ or Cmd-
https://bugs.webkit.org/show_bug.cgi?id=247638
rdar://102385998

Reviewed by Aditya Keerthi.

This PR just removes some `IOS_FAMILY` guards so that macOS behavior matches iOS.

* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::nodeHeight):
(WebCore::RenderFileUploadControl::maxFilenameWidth const):
(WebCore::RenderFileUploadControl::paintControl):

Canonical link: https://commits.webkit.org/258277@main
